mlir/lib/Dialect/GPU/IR/GPUDialect.cpp

```diff
@@ -106,6 +106,20 @@ int64_t GPUThreadMappingAttr::getRelativeIndex() const {
              : getMappingId();
 }
 
+int64_t GPULaneMappingAttr::getMappingId() const {
+  return static_cast<int64_t>(getLane());
+}
+
+bool GPULaneMappingAttr::isLinearMapping() const {
+  return getMappingId() >= static_cast<int64_t>(MappingId::LinearDim0);
+}
+
+int64_t GPULaneMappingAttr::getRelativeIndex() const {
```

Comment:
I'm mildly confused. Where are the declarations of these methods?

Reply:
in `DeviceMappingAttrInterface` @ mlir/include/mlir/Dialect/SCF/IR/DeviceMappingInterface.td